app/controllers/users_controller.rb

```diff
@@ -2,7 +2,8 @@ class UsersController < ApplicationController
   # before_action :authenticate_user!, except: [:show]
 
   def show
-    @user = User.find(params[:id])
+    @user = current_user
+    @regapp = current_user.regapps
```

Comment:
This line does the same thing as the line below it. So you have a singularly named instance var that contains an array of objects. This can make things very confusing later on.